nms/app/context/FEGGatewayContext.tsx

```diff
@@ -143,16 +145,24 @@ async function setGatewayState(params: GatewayStateParams) {
     );
     setFegGatewaysHealthStatus(newFegGatewaysHealthStatus);
   } else {
-    await MagmaAPI.federationGateways.fegNetworkIdGatewaysGatewayIdDelete({
-      networkId: networkId,
-      gatewayId: key,
-    });
-    const newFegGateways = {...fegGateways};
-    const newFegGatewaysHealthStatus = {...fegGatewaysHealthStatus};
-    delete newFegGateways[key];
-    delete newFegGatewaysHealthStatus[key];
-    setFegGateways(newFegGateways);
-    setFegGatewaysHealthStatus(newFegGatewaysHealthStatus);
+    if (tierId !== undefined && tierId !== '') {
+      await MagmaAPI.lteGateways.lteNetworkIdGatewaysGatewayIdTierPut({
+        networkId,
+        gatewayId: key,
+        tierId: JSON.stringify(`"${tierId}"`),
+      });
+    } else {
+      await MagmaAPI.federationGateways.fegNetworkIdGatewaysGatewayIdDelete({
```

Comment:
So if I pass in an empty string as `tierId` the gateway gets deleted 🧐
These multipurpose methods as just asking for things to go wrong. See comment above.